applications/nrf_desktop/doc/hid_provider_system_ctrl.rst

```diff
@@ -23,117 +23,16 @@ Module events
-Before a connection with the HID host is established, the provider enqueues the HID usage ID and keypress state (press or release) in the HID event queue.
-Once the connection is established, the elements of the queue are replayed one after the other to the host, in a sequence of consecutive HID reports.
-On the HID state request, the module pulls an element from the queue, updates the tracked state of pressed keys, and submits a :c:struct:`hid_report_event` to provide the HID input report to the HID subscriber.
-The subsequent requests lead to providing subsequent keypresses as HID report events.
-All key state changes still go through the HID event queue until the queue is empty.
-
-Once the queue is empty, a key state change results in an instant update of the tracked state of pressed keys.
-If the state of pressed keys changes, the module calls the :c:member:`hid_state_api.trigger_report_send` callback to notify the :ref:`nrf_desktop_hid_state` about the new data.
-The module also remembers that the HID subscriber needs to be updated.
-
-.. note::
-   The module tracks and reports a single HID system control usage ID at a time.
-   If multiple keys are simultaneously pressed, the module reports only the first pressed key.
-   Other keys are ignored.
-
-Discarding queued events
-------------------------
+.. |config_consumer_system_crtl| replace:: :ref:`CONFIG_DESKTOP_HID_REPORT_PROVIDER_SYSTEM_CTRL <config_desktop_app_options>`
 
-While key state changes go through the event queue and there is no space for a new input event, the module tries to free space by discarding the oldest event in the queue.
-Events stored in the queue are automatically discarded after the period defined by the :ref:`CONFIG_DESKTOP_HID_REPORT_PROVIDER_SYSTEM_CTRL_KEYPRESS_EXPIRATION <config_desktop_app_options>` option.
+.. |config_consumer_system_crtl_support| replace:: :ref:`CONFIG_DESKTOP_HID_REPORT_SYSTEM_CTRL_SUPPORT <config_desktop_app_options>`
 
-When discarding an event from the queue, the module checks if the key associated with the event is pressed.
-This is to avoid missing key releases for earlier key presses when the keys from the queue are replayed to the host.
-If a key release is missed, the host could stay with a key that is permanently pressed.
-The discarding mechanism ensures that the host will always receive the correct key sequence.
+.. |config_consumer_system_crtl_alt| replace:: :ref:`CONFIG_DESKTOP_HID_REPORT_PROVIDER_SYSTEM_CTRL_ALT <config_desktop_app_options>`
 
-.. note::
-    The module can only discard an event if the event does not overlap any button that was pressed but not released, or if the button itself is pressed.
-    The event is released only when the following conditions are met:
+.. |config_consumer_system_crtl_event_queue_size| replace:: :ref:`CONFIG_DESKTOP_HID_REPORT_PROVIDER_SYSTEM_CTRL_EVENT_QUEUE_SIZE <config_desktop_app_options>`
```

Comment:
Can we simply replace `CONSUMER`/ `SYSTEM` as part of Kconfig option names? Or do we need to apply the replacement for whole Kconfig options ?

Reply:
Checked replacing the only the text and it doesn't work. I think the whole Kconfig should be replaced.